Content.Shared/Cabinet/SealingCabinetSystem.cs

```diff
@@ -0,0 +1,47 @@
+using Content.Shared.Emag.Systems;
+using Content.Shared.Nutrition.Components;
+using Content.Shared.Nutrition.EntitySystems;
+using Content.Shared.Popups;
+
+namespace Content.Shared.Cabinet;
+
+public sealed class SealingCabinetSystem : EntitySystem
+{
+    [Dependency] private readonly ItemCabinetSystem _cabinet = default!;
+    [Dependency] private readonly OpenableSystem _openable = default!;
+    [Dependency] private readonly SharedPopupSystem _popup = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<SealingCabinetComponent, OpenableOpenAttemptEvent>(OnOpenAttempt);
+        SubscribeLocalEvent<SealingCabinetComponent, GotEmaggedEvent>(OnEmagged);
+    }
+
+    private void OnOpenAttempt(Entity<SealingCabinetComponent> ent, ref OpenableOpenAttemptEvent args)
+    {
+        var cabinet = (ent, Comp<ItemCabinetComponent>(ent));
+        if (!_cabinet.HasItem(cabinet))
+            return;
+
```

Comment:
Comp will throw if they don't have the component so better to trycomp or omit entirely.

Reply:
TryComp and logging an informative error on failure is probably preferable to throwing an exception.

Reply:
Any component can be added or removed at runtime, VV is indicative of this, and throwing should be avoided at all costs if it's expected.